lib/iex/test/iex/autocomplete_test.exs

```diff
@@ -162,4 +162,36 @@ defmodule IEx.AutocompleteTest do
     assert expand('EList') == {:yes, '.', []}
     assert expand('EList.map') == {:yes, [], ['map/2', 'mapfoldl/3', 'mapfoldr/3']}
   end
+
+  test "completion for functions added when a module is reloaded" do
+    filename = "sample.ex"
+    content = """
+    defmodule Sample do
+    def foo(), do: 0
+    end
+    """
+
+    reloaded_content = """
+    defmodule Sample do
+    def foo(), do: 0
+    def foobar(), do: 0
+    end
+    """
+    File.write!(filename, content)
+    compiler_options = :elixir_config.get(:compiler_options)
```

Comment:
Oh, `capture_io` would have indeed been better.

The purpose of the test is to mimic `r/1` helper, as it is what I was trying to fix.
The autocomplete was already working fine when defining module inline, which is why I used `Code.load_file` as this is what `r/1` uses to reload the module.